pkg/setup/vscode.go

```diff
@@ -0,0 +1,178 @@
+package setup
+
+import (
+	"encoding/json"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"os"
+	"path"
+	"regexp"
+)
+
+const (
+	vscodeDirName            = ".vscode"
+	vscodeLaunchFileName     = "launch.json"
+	gebugLaunchName          = "Gebug"
+	defaultVsCodeConfVersion = "0.2.0"
+)
+
+type VsCode struct {
+	baseIde
+}
+
+func NewVsCode(workDir string, port int) *VsCode {
+	return &VsCode{baseIde{
+		workDir:      workDir,
+		debuggerPort: port,
+	}}
+}
+func (v VsCode) Detected() (bool, error) {
+	return v.detected(vscodeDirName)
+}
+
+func (v VsCode) GebugInstalled() (bool, error) {
+	detected, err := v.Detected()
+	if err != nil {
+		return false, err
+	}
+	if !detected {
+		return false, errors.New("vscode was not detected in this workspace")
+	}
+	launchConfigPath := path.Join(v.workDir, vscodeDirName, vscodeLaunchFileName)
+	launchContent, err := afero.ReadFile(AppFs, launchConfigPath)
+	if err != nil {
+		return false, errors.WithMessage(err, "read vscode launch.json file")
+	}
+	installed, err := v.installedInLaunchConfig(launchContent)
+	if err != nil {
+		return false, errors.WithMessage(err, "check if installed in launch.json")
+	}
+
+	return installed, nil
+}
+
+func (v VsCode) Enable() error {
+	return v.setEnabled(true)
+}
+
+func (v VsCode) Disable() error {
+	return v.setEnabled(false)
+}
+
+func (v VsCode) installedInLaunchConfig(in []byte) (bool, error) {
+	var launchConfig = struct {
+		Version        string
+		Configurations []struct {
+			Name string
+		}
+	}{}
+
+	err := json.Unmarshal(v.removeComments(in), &launchConfig)
+	if err != nil {
+		return false, errors.WithMessage(err, "unmarshal no comments input")
+	}
+	for _, c := range launchConfig.Configurations {
+		if c.Name != gebugLaunchName {
+			continue
+		}
+		return true, nil
+	}
+	return false, nil
+}
+
+func (v VsCode) removeComments(in [] byte) []byte {
+	re := regexp.MustCompile("(?s)//.*?\n|/\\*.*?\\*/")
```

Comment:
Are we ok with it panicking here? maybe we want an error instead?
what do you think?

Reply:
It's a static pattern and handling error here will make the code too complicated without a good reason